c/parallel/src/scan.cu

```diff
@@ -0,0 +1,538 @@
+      ltoir_list_append({input_it.dereference.ltoir, input_it.dereference.ltoir_size});
+    }
+    if (cccl_iterator_kind_t::iterator == output_it.type)
+    {
+      ltoir_list_append({output_it.advance.ltoir, output_it.advance.ltoir_size});
+      ltoir_list_append({output_it.dereference.ltoir, output_it.dereference.ltoir_size});
+    }
+
+    nvrtc_link_result result =
+      make_nvrtc_command_list()
+        .add_program(nvrtc_translation_unit{src.c_str(), name})
+        .add_expression({init_kernel_name})
+        .add_expression({scan_kernel_name})
+        .compile_program({args, num_args})
+        .get_name({init_kernel_name, init_kernel_lowered_name})
+        .get_name({scan_kernel_name, scan_kernel_lowered_name})
+        .cleanup_program()
+        .add_link_list(ltoir_list)
+        .finalize_program(num_lto_args, lopts);
+
+    cuLibraryLoadData(&build->library, result.data.get(), nullptr, nullptr, 0, nullptr, nullptr, 0);
+    check(cuLibraryGetKernel(&build->init_kernel, build->library, init_kernel_lowered_name.c_str()));
+    check(cuLibraryGetKernel(&build->scan_kernel, build->library, scan_kernel_lowered_name.c_str()));
+
+    constexpr size_t num_ptx_args      = 7;
+    const char* ptx_args[num_ptx_args] = {
+      arch.c_str(), cub_path, thrust_path, libcudacxx_path, ctk_path, "-rdc=true", "-dlto"};
+    constexpr size_t num_ptx_lto_args       = 3;
+    const char* ptx_lopts[num_ptx_lto_args] = {"-lto", arch.c_str(), "-ptx"};
+
+    std::string ptx_src = std::format(
+      "#include <cub/agent/single_pass_scan_operators.cuh>\n"
+      "#include <cub/util_type.cuh>\n"
+      "struct __align__({1}) storage_t {{\n"
+      "  char data[{0}];\n"
+      "}};\n"
+      "__device__ size_t description_bytes_per_tile = cub::ScanTileState<{2}>::description_bytes_per_tile;\n"
+      "__device__ size_t payload_bytes_per_tile = cub::ScanTileState<{2}>::payload_bytes_per_tile;\n",
+      accum_t.size,
+      accum_t.alignment,
+      accum_cpp);
+    auto compile_result =
+      make_nvrtc_command_list()
+        .add_program(nvrtc_translation_unit{ptx_src.c_str(), "tile_state_info"})
+        .compile_program({ptx_args, num_ptx_args})
+        .cleanup_program()
+        .finalize_program(num_ptx_lto_args, ptx_lopts);
+    auto ptx_code = compile_result.data.get();
+
+    size_t description_bytes_per_tile;
+    size_t payload_bytes_per_tile;
+    auto maybe_description_bytes_per_tile = scan::find_size_t(ptx_code, "description_bytes_per_tile");
+    if (maybe_description_bytes_per_tile)
+    {
+      description_bytes_per_tile = maybe_description_bytes_per_tile.value();
+    }
+    else
+    {
+      throw std::runtime_error("Failed to find description_bytes_per_tile in PTX");
+    }
+    payload_bytes_per_tile = scan::find_size_t(ptx_code, "payload_bytes_per_tile").value_or(0);
+
+    auto tile_state = std::make_unique<scan::scan_tile_state>(description_bytes_per_tile, payload_bytes_per_tile);
```

Comment:
question: why was this necessary? It looks like we always return a copy of tile state from `TileState`. Instead of tracking heap allocations, can we just store:
```c++
  size_t description_bytes_per_tile;
  size_t payload_bytes_per_tile;
```
in the build object by value directly? Then, in the `TileState` we'd just construct tile state as `return scan_tile_state{build.description_bytes_per_tile, build.payload_bytes_per_tile}` and save one memory allocation.

Reply:
Fixed in https://github.com/NVIDIA/cccl/pull/3462/commits/7a83fbc990af0395fde43721a72f31d931d8f0fc.